src/Microsoft.Diagnostics.Monitoring/DiagnosticsMonitor.cs

```diff
@@ -19,359 +19,125 @@ namespace Microsoft.Diagnostics.Monitoring
 {
     public sealed class DiagnosticsMonitor : IAsyncDisposable
```

Comment:
What if we renamed this?
```suggestion
    public sealed class EventPipePipeline : IAsyncDisposable
```

I've also got a suggestion for refactoring but even if you agree don't feel obliged to do it within this PR:
In the current design it feels like DiagnosticsMonitor and DiagnosticsEventPipeProcessor each play the role of pipeline API in different situations. In the cases that emit nettrace directly DiagnosticsMonitor does it but in the case that we need further processing DiagnosticsEventPipeProcessor becomes the outer API and DiagnosticsMonitor becomes a helper. Ideally I'd like to keep the API as normalized as possible between the cases which leads me towards two possibilities:
1. Both of these types implement some common IPipeline interface and implement it independently or with a common base class
2. We invert control so that DiagnosticsMonitor is the exposed pipeline API always and EventPipeProcessor is passed in as a constructor argument. After DiagnosticsMonitor sets up the incoming nettrace data stream it would then provide that stream to the processor and let it do whatever it wants with it. A no-op EventPipeProcessor would copy the raw data to the output stream, or a logs/metrics processor would parse it with TraceEvent and serialize a new output stream. There is also the possibility to merge the processor with the existing MonitoringSourceConfiguration object so you might end up with usage that looks like this:
````
EventPipePipeline a = new EventPipePipeline(new NetTraceConfiguration(eventPipeProviders, outputStream));
EventPipePipeline b = new EventPipePipeline(new LogStreamConfiguration(outputStream));
EventPipePipeline c = new EventPipePipeline(new MetricStreamConfiguration(intervalSeconds:60, outputStream));
````

My guess is that (2) will be simpler but either could work.

Reply:
My initial thinking was that the DiagnosticsMonitor (now poorly named) would just be folded into DiagnosticsEventPipeProcessor. Depending on the output configuration, the pipeline would output raw nettrace data or other results. 
Some other thoughts:
- I think keeping these as the same pipeline makes the most sense. 
- I think we should keep configuration decoupled from output. For example, you can specify the same LogStreamConfiguration, but choose to get nettrace or json.

Reply:
Folding them together into a single type is fine with me and fair point on the Logging configuration. Sounds good!